bson/codec_options.pyi

```diff
@@ -22,7 +22,8 @@ you get the error: "TypeError: 'type' object is not subscriptable".
 import datetime
 import abc
 import enum
-from typing import Tuple, Generic, Optional, Mapping, Any, TypeVar, Type, Dict, Iterable, Tuple, MutableMapping, Callable, Union
+from typing import Tuple, Generic, Optional, Mapping, Any, TypeVar, Type, Dict, Iterable, Tuple, Callable, Union
+from pymongo.typings import _DocumentType, _DocumentTypeArg
```

Comment:
Let's reverse this so that _DocumentType and _DocumentTypeArg are defined in the bson package. It's odd to have bson import from pymongo, even if it's for typing.

Reply:
I created `bson/typings.py` and shuffled things around